libs/s25main/mapGenerator/Random.h

```diff
@@ -22,14 +22,19 @@
 #include "mapGenerator/MapSettings.h"
 #include "mapGenerator/Map.h"
 
+namespace rttr {
+namespace mapGenerator {
+
 class Random
 {
 private:
-    RandomUtility rnd_;
+    algorithm::RandomUtility rnd_;
 
 public:
-    Random(RandomUtility& rnd) : rnd_(rnd) {}
+    Random(algorithm::RandomUtility& rnd) : rnd_(rnd) {}
```

Comment:
You are copying the instance. Is that on purpose? It looks wrong because if you are using the original somewhere else you'll get the same random numbers (guessing what the class does here)

Reply:
**Fixed**
There was no need copying the `RandomUtility` instance - also it lead to weird behavior as you said.